Makefile.am

```diff
@@ -6615,7 +6614,7 @@ valgrind-tests: $(TESTS)
 		if $(LIBTOOL) --mode=execute file $$f | grep -q shell; then \
 		echo -e "$${x}Skipping non-binary $$f"; else \
 		echo -e "$${x}Running $$f"; \
-		$(AM_TESTS_ENVIRONMENT) $(LIBTOOL) --mode=execute valgrind -q --leak-check=full --max-stackframe=5242880 --error-exitcode=55 $(builddir)/$$f ; fi; \
+		$(LIBTOOL) --mode=execute valgrind -q --leak-check=full --max-stackframe=5242880 --error-exitcode=55 $(builddir)/$$f ; fi; \
```

Comment:
Yes, this was an explicit fix, we also need it for the other variables -- otherwise we test against the *installed* systemd binaries, but we want those from the build tree. @keszybz, I force-pushed your commit to only keep the first hunk, and revert the invocation of valgrind (i. e. the second hunk).